eole/modules/multi_headed_attn.py

```diff
@@ -614,6 +614,16 @@ def forward(
                         key = key[:, :, 1:, :]
                         value = value[:, :, 1:, :]
 
+                if step == 0:
+                    key_pad_mask = self.layer_cache[1].get("key_pad_mask", None)
+                    if key_pad_mask is not None:
+                        x = key_pad_mask.expand(
+                            -1, self.heads // self.parallel_gpu, -1
```

Comment:
@vince62s @l-k-11235 this seems to break llama3 inference (and probably others).
Replacing `self.heads // self.parallel_gpu` with `value.size(1)` would probably fix it, but not sure it'll cover all cases.

```
Traceback (most recent call last):
  File "/usr/local/bin/eole", line 33, in <module>
    sys.exit(load_entry_point('EOLE', 'console_scripts', 'eole')())
  File "/work/eole/bin/main.py", line 39, in main
    bin_cls.run(args)
  File "/work/eole/bin/run/predict.py", line 42, in run
    predict(config)
  File "/work/eole/bin/run/predict.py", line 18, in predict
    _, _, _ = engine.infer_file()
  File "/work/eole/inference_engine.py", line 35, in infer_file
    scores, estims, preds = self._predict(infer_iter)
  File "/work/eole/inference_engine.py", line 159, in _predict
    scores, estims, preds = self.predictor._predict(
  File "/work/eole/predict/inference.py", line 454, in _predict
    batch_data = self.predict_batch(batch, attn_debug)
  File "/work/eole/predict/generator.py", line 71, in predict_batch
    return self._predict_batch_with_strategy(batch, decode_strategy)
  File "/work/eole/predict/generator.py", line 135, in _predict_batch_with_strategy
    log_probs, attn = self._decode_and_generate(
  File "/work/eole/predict/inference.py", line 664, in _decode_and_generate
    dec_out, dec_attn = self.model.decoder(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "/work/eole/decoders/transformer_lm_decoder.py", line 137, in forward
    emb, attn, _ = layer(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "/work/eole/decoders/transformer_base.py", line 76, in forward
    layer_out, attns = self._forward(*args, **kwargs)
  File "/work/eole/decoders/transformer_lm_decoder.py", line 56, in _forward
    attn_output, attns = self.self_attn(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "/work/eole/modules/multi_headed_attn.py", line 619, in forward
    value = value.masked_fill(x, 0)
RuntimeError: The size of tensor a (32) must match the size of tensor b (8) at non-singleton dimension 1
```